src/bodywork/stage_execution.py

```diff
@@ -57,12 +68,23 @@ def run_stage(
         stage = project_config.stages[stage_name]
         if stage.requirements:
             _install_python_requirements(stage.requirements)
-        run(
-            ["python", stage.executable_module, *stage.args],
-            check=True,
-            cwd=stage.executable_module_path.parent,
-            encoding="utf-8",
-        )
+        executable_type = _infer_executable_type(stage.executable_module)
+        if executable_type is ExecutableType.JUPYTER_NB:
+            notebook = nbformat.read(
+                stage.executable_module_path, as_version=nbformat.NO_CONVERT
+            )
+            nb_runner = ExecutePreprocessor()
+            nb_runner.preprocess(
+                notebook,
+                {"metadata": {"path": stage.executable_module_path.parent}},
+            )
+        else:
+            run(
+                ["python", stage.executable_module, *stage.args],
+                check=True,
+                cwd=stage.executable_module_path.parent,
+                encoding="utf-8",
+            )
```

Comment:
We need to think if we are going to add to the list of executables files that we will have in the future. If we do then the OO approach of a class with interface containing an `execute` method for each type would be nicer and extensible (open/closed principle).